csp_billing_adapter_microsoft/plugin.py

```diff
@@ -71,6 +70,102 @@ def meter_billing(
     is attempted 3 times before re-raising the exception to
     calling scope.
     """
+    metered_billing_data = {"request": []}
+    status = {}
+
+    for dimension_name, quantity in dimensions.items():
+        if quantity == 0:
+            log.info(
+                'A "0" value was reported for %s, skipping meter billing',
+                dimension_name
+            )
+            continue
+
+        # Setup request body for the usage event API
+        metered_billing_data["request"].append(
+            {
+                'resourceUri': os.environ['EXTENSION_RESOURCE_ID'],
+                'quantity': quantity,
+                'dimension': dimension_name,
+                'effectiveStartTime': str(timestamp.astimezone(timezone.utc)),
+                'planId': os.environ['PLAN_ID']
+            }
+        )
+
+    if metered_billing_data["request"]:
+        # Setup request header for the usage event API
+        metered_billing_header = {
+            'Content-type': 'application/json',
+            'x-ms-correlationid': str(uuid.uuid4()),
+            'authorization': _get_msi_token()
+        }
+
+        # Make request to the usage event API
+        data_request = urllib.request.Request(
+            'https://marketplaceapi.microsoft.com/api/batchUsageEvent'
+            '?api-version=2018-08-31',
+            data=json.dumps(metered_billing_data).encode(),
```

Comment:
Should the encode here be specifying an encoding type? The response extraction below is decoding as utf-8...

Reply:
it should thanks for catching this.

Reply:
Thanks for catching that, I added it in